typeform/typeform.py

```diff
@@ -0,0 +1,182 @@
+import time
+import copy
+import json
+import panoply
+import urllib2
+import dateparser
+
+
+FETCH_LIMIT = 500
+DESTINATION = 'typeform'
+DESTINATION_POSTFIX = '_{__table}'
+BASE_URL = 'https://api.typeform.com/v1'
+
+
+class Typeform(panoply.DataSource):
+
+    # constructor
+    def __init__(self, source, options):
+        super(Typeform, self).__init__(source, options)
+
+        if 'destination' not in source:
+            source['destination'] = DESTINATION
+
+        # append the destination postfix, which represent the form name
+        source['destination'] += DESTINATION_POSTFIX
+
+        # configure incremental pulling
+        self._incval = None
+        if (source.get('inckey')):
+            incval = source.get('incval')
+            self._incval = (incval if incval else
+                source.get('lastTimeSucceed') or source.get('lastRuntime'))
+
+            # validate that's the incremental value is a valid
+            # date that could be parsed to a Unix timestamp before
+            # including it in the process
+            self._incval = getTimestamp(self._incval)
+
+
+        forms = source.get('forms')
+        forms = forms if forms else []
```

Comment:
this is true but I got burnt by it when I tried to debug `Librato` it had a line like this 
`forms = source.get('forms', [])` (maybe it wasn't forms but something like this)
and then later in the code:
`blabla= map(... , forms)`
and this line failed with `forms is not iterable` because source did had forms defined but it was defined as undefined (/ None).

so in short my problem with get is that sometimes the key is defined but the value is undefined (None)